EIPS/eip-5164.md

```diff
@@ -123,22 +132,58 @@ interface CrossChainRelayer {
 
 The `CrossChainExecutor` executes relayed calls. Developers must implement a `CrossChainExecutor` in order to execute messages on the receiving chain.
 
-#### Authentication
+The `CrossChainExecutor` will execute a nonce only once, but may execute nonces in any order. This specification makes no ordering guarantees, because messages may travel async through the transport layer.
 
-`CrossChainRelayer`s SHOULD authenticate that the call has been performed by the bridge transport layer.
+#### Execution
 
-`CrossChainRelayer`s SHOULD use [EIP-2771](./eip-2771.md) for authentication and append the address of the Transaction Signer (20 bytes of data) on the origin chain to the end of the call data being executed.
+`CrossChainExecutor`s SHOULD authenticate that the call has been performed by the bridge transport layer.
+
+`CrossChainExecutor`s MUST NOT execute a batch of calls more than once.
+
+**Calldata**
+
+`CrossChainExecutor`s MUST append the ABI-packed (`nonce`, `sender`) to the calldata for each call being executed. It allows the receiver of the call to check the true `sender` of the transaction and use the `nonce` to apply any transaction ordering logic.
+
+```solidity
+interface CrossChainExecutor {
+  bytes calldata = encode(Call.data, nonce, sender);
+}
```

Comment:
I didn't want to specify if the developer should use `abi.encode` or `abi.encodePacked` so that why I've used the more generic `encode` but it's true that this it not the proper Solidity syntax.
Any recommendation on how to handle this case? Thank you.

Reply:
```suggestion
interface CrossChainExecutor {
  bytes calldata = abi.encode(Call.data, nonce, sender); // Can also use abi.encodePacked
}
```

Reply:
I'm more confused by the variable defined in an interface. I don't think that's valid Solidity.